packages/cli/core/src/commands/atomic/cmp.ts

```diff
@@ -0,0 +1,65 @@
+import {CLICommand} from '@coveo/cli-commons/command/cliCommand';
+import {UnknownError} from '@coveo/cli-commons/errors/unknownError';
+import {Flags} from '@oclif/core';
+import inquirer from 'inquirer';
+import {appendCmdIfWindows} from '../../lib/utils/os';
+import {spawnProcess} from '../../lib/utils/process';
+
+export default class AtomicInit extends CLICommand {
+  public static description =
+    'Scaffold a new custom component in your Coveo Atomic Component Library';
+
+  public static examples = [
+    '<%= config.bin %> <%= command.id %> --type=page myAwesomeComponent',
+    '<%= config.bin %> <%= command.id %> --type=result mySuperResultComponent',
+  ];
+
+  public static flags = {
+    type: Flags.string({
+      description: 'the kind of component to initialize',
+      options: ['page', 'result'],
+    }),
+  };
+
+  public static args = [
+    {name: 'name', description: 'the name of your component', required: true},
+  ];
+
+  public async run(): Promise<void> {
+    const {initializer, name} = await this.getSpawnOptions();
+
+    const cliArgs = ['init', initializer, name];
+    await spawnProcess(appendCmdIfWindows`npm`, cliArgs);
```

Comment:
There seems to be a decent amount of instances of `spawnProcess` being used to spawn NPM, and it may not be immediately intuitive that we need to append the CMD if we make other commands which spawn NPM. I think it may be worth extracting a `spawnNPM` util, or more specific functions like `npmInit`, `npmInstall`, etc.

Reply:
Could be nice, I'll keep that out-of-scope of this PR tho, there's already a lot.

Reply:
CDX-1372